### PR TITLE
Wagtail 6.1 upgrade

### PR DESCRIPTION
**Pull request by engineervix**
When merged this will upgrade wagtail to v6.1

[Support Ticket](https://torchbox.monday.com/boards/1318157635/pulses/1529770547)

[Wagtail 6.1 release notes](https://docs.wagtail.org/en/stable/releases/6.1.html)

In addition to upgrading wagtail to version 6.1

* `beautifulsoup4` and `html5lib` are explicitly declared in `pyproject.toml`, as they are used in various places on the codebase. This was driven by [the removal of `html5lib` dependency](https://docs.wagtail.org/en/latest/releases/6.1.html#removal-of-html5lib-dependency) in Wagtail 6.1
* Reorganise python dependencies for better readability (524d3d81)
* several outdated dependencies are updated (see details below)

<details>
  <summary>Updated dependencies</summary>
  
| Commit     | Description                                          |
|------------|------------------------------------------------------|
| fd943d39   | Update django (4.2.11 -> 4.2.13)           |
| fca7b16f   | Update urllib3 (1.26.13 -> 1.26.18) and ensure "urlib3<2"|
| 97fb70a9   | Update werkzeug (3.0.2 -> 3.0.3)                       |
| 16f3b93d   | Update sqlparse (0.4.3 -> 0.5.0)                       |
| 0276afb4   | Update requests (2.31.0 -> 2.32.3)                    |
| ecce4300   | Update redis (4.4.0 -> 5.0.6)                          |
| 01b6900f   | Update pillow (9.3.0 -> 10.3.0)                        |
| b3b4f8f1   | Update paramiko (2.12.0 -> 3.4.0)                      |
| 210b3695   | Update jinja2 (3.1.2 -> 3.1.4)                         |
| e42f2e40   | Update idna (3.4 -> 3.7)                               |
| 0ccd9f87   | Update cryptography (38.0.4 -> 42.0.8)                 |
| a8c0ca0a   | Update certifi (2022.12.7 -> 2024.6.2)                 |
| 8532bcd4c82ece41c7eb223f5799900261ed05f2   | Update django-upgrade (1.16.0 -> 1.18.0)   |
| 66f95d0ba8936712815e965dc2b2751e2d183161  | Update pymdown-extensions (10.8 -> 10.8.1) |
| ceeb71a1fa2c76e5fffb24404bb8537e3e0aa4dd   | Update wagtail-factories (4.1.0 -> 4.2.1)  |
| 0910ff907f007e78c343b48a0af1e2cbd08dcb28   | Update lxml (5.2.1 -> 5.2.2)               |
| 0257256b5be7c0ecbc7ec8e11bac5ed8c8674e31  | Update coverage (7.4.4 -> 7.5.3)           |

</details>

## Wagtail 6.1 upgrade check list

Add a link to the commit or file for each consideration if you made any code changes.

## Code reviewing a consideration

Use the tick box to indicate a consideration has been code reviewed as OK

### Changes affecting wagtail customizations

- [x] [Changes to submissionslistview class](https://docs.wagtail.org/en/stable/releases/6.1.html##changes-to-submissionslistview-class)
- [x] [Register_user_listing_buttons hook signature changed](https://docs.wagtail.org/en/stable/releases/6.1.html##register-user-listing-buttons-hook-signature-changed) 
- [x] [Password_required_template has changed to wagtail_password_required_template](https://docs.wagtail.org/en/stable/releases/6.1.html##password-required-template-has-changed-to-wagtail-password-required-template) | (693ef95d)
- [x] [Document_password_required_template has changed to wagtaildocs_password_required_template](https://docs.wagtail.org/en/stable/releases/6.1.html##document-password-required-template-has-changed-to-wagtaildocs-password-required-template)

### Changes to undocumented internals

- [x] [Removal of `html5lib` dependency](https://docs.wagtail.org/en/latest/releases/6.1.html#removal-of-html5lib-dependency) | (77979058) 
- [x] [Deprecation of user_listing_buttons template tag](https://docs.wagtail.org/en/stable/releases/6.1.html##deprecation-of-user-listing-buttons-template-tag) | (2645c204)
- [x] [Deprecation of wagtailusers_groups:users url pattern](https://docs.wagtail.org/en/stable/releases/6.1.html##deprecation-of-wagtailusers-groups-users-url-pattern)
- [x] [Deprecation of window.chooserurls within draftail choosers](https://docs.wagtail.org/en/stable/releases/6.1.html##deprecation-of-window-chooserurls-within-draftail-choosers)
- [x] [Deprecation of window.initblockwidget to initialize a streamfield block](https://docs.wagtail.org/en/stable/releases/6.1.html##deprecation-of-window-initblockwidget-to-initialize-a-streamfield-block)
- [x] [Removal of jquery from base client-side widget and boundwidget classes](https://docs.wagtail.org/en/stable/releases/6.1.html##removal-of-jquery-from-base-client-side-widget-and-boundwidget-classes)
- [x] [Window.urlify deprecated](https://docs.wagtail.org/en/stable/releases/6.1.html##window-urlify-deprecated)
- [x] [Window.xregexp polyfill removed](https://docs.wagtail.org/en/stable/releases/6.1.html##window-xregexp-polyfill-removed)

### CODE REVIEW

File: docs/upgrading.md [hunk 1]
```diff
@@ -154,10 +154,25 @@ Additionally there are some Custom Panels which help to add the `permission` par
 
 The following templates are overridden and should be checked for changes when upgrading Wagtail:
 
-Last checked against Wagtail version: 6.0
+Last checked against Wagtail version: 6.1
 
 - `rca/account_management/templates/wagtailadmin/base.html`
-- `rca/users/templates/wagtailusers/users/list.html`
+- ~~`rca/users/templates/wagtailusers/users/list.html`~~ This template was deleted in 2645c204425b8fa3409a110f46b2822a1953fe49 because as of Wagtail 6.1, [it's no longer used](https://github.com/wagtail/wagtail/commit/7b1644eb37b6b6cf7800276acf9abef5254fc096). Please note that the `user_listing_buttons` template tag was used in this template, and it has since been [deprecated](https://docs.wagtail.org/en/latest/releases/6.1.html#deprecation-of-user-listing-buttons-template-tag).
```
Comment (nickmoreton) [inline docs/upgrading.md:160]: **Suggestion:** please can this change here be highlighted to the client to test the specific functionality that should be present here. Either as a special note on the upgrade ticket or via the testing documentation we send to the client.

I tested it locally and it seems to be OK.

**Review by nickmoreton — APPROVED** [review]
I think we need to make sure the client is aware of the changes to the user list so extra focus can be placed on testing that functionality.

**Comment by engineervix** [comment]
**NOTE**: I've added two commits that update django to 4.2.14 and wagtail to 6.1.3 respectively.

<details><summary>More details on the Wagtail update</summary>

- https://docs.wagtail.org/en/latest/releases/6.1.3.html
- https://github.com/wagtail/wagtail/security/advisories/GHSA-jmp3-39vp-fwg8

</details>

<details><summary>More details on the Django update</summary>

- https://docs.djangoproject.com/en/5.0/releases/4.2.14/
- https://www.djangoproject.com/weblog/2024/jul/09/security-releases/

</details>

**Review by nickmoreton — APPROVED** [review]
Approving the security fixes

**Pull request merged**